net/routing/src/etc/inc/plugins.inc.d/routing.inc

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+    Copyright (C) 2017 Fabian Franz
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    1. Redistributions of source code must retain the above copyright notice,
+       this list of conditions and the following disclaimer.
+
+    2. Redistributions in binary form must reproduce the above copyright
+       notice, this list of conditions and the following disclaimer in the
+       documentation and/or other materials provided with the distribution.
+
+    THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES,
+    INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+    AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+    OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+    SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+    CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+
+function routing_services()
+{
+    global $config;
+
+    $services = array();
+
+    if (isset($config['OPNsense']['routing']['general']['enabled']) && $config['OPNsense']['routing']['general']['enabled'] == 1) {
+        $services[] = array(
+            'description' => gettext('Routing deamon to add support of various routing protocols.'),
+            'configd' => array(
+                'restart' => array('quagga restart'),
+                'start' => array('quagga start'),
+                'stop' => array('quagga stop'),
+            ),
+            'name' => 'Routing',
```

Comment:
yes, this is a plugin that depends on a specific daemon. core services could be called "routing", but it could also be highly confusing for users. Plus the next plugin based on OpenBGPD or OpenOSPFD would have to go a different route in naming, we can't use "Routing2" here. Same rules for all potential plugins.